src/components/app/details/appDetails/AppDetails.components.tsx

```diff
@@ -0,0 +1,127 @@
+import { generatePath, useHistory, useParams, useRouteMatch } from 'react-router-dom'
+import { GroupBase } from 'react-select'
+
+import {
+    AppEnvironment,
+    EnvAppsMetaDTO,
+    SelectPicker,
+    SelectPickerOptionType,
+    SelectPickerProps,
+} from '@devtron-labs/devtron-fe-common-lib'
+
+import { AppEnvSelectorProps } from './appDetails.type'
+import { getEnvOptions } from './utils'
+
+const AppEnvDropdown = ({
+    name,
+    options,
+    onChange,
+    value,
+}: Pick<SelectPickerProps, 'name' | 'options' | 'onChange' | 'value'>) => (
+    <>
+        <div style={{ width: 'clamp( 100px, 30%, 100px )', height: '100%', position: 'relative' }}>
+            <svg
+                viewBox="0 0 200 40"
+                preserveAspectRatio="none"
+                style={{ width: '100%', height: '100%', display: 'flex' }}
+            >
+                <path d="M0 20 L200 20 Z" strokeWidth="1" stroke="var(--B500)" />
+                <path d="M0 10 L0, 30" strokeWidth="2" stroke="var(--B500)" />
+            </svg>
+            <div
+                className="bcb-5 br-10 cn-0 pl-8 pr-8"
+                style={{ position: 'absolute', top: '50%', left: '50%', transform: 'translate(-50%, -50%)' }}
+            >
+                {name.slice(0, 3).toUpperCase()}
+            </div>
+        </div>
+        <div data-testid="app-deployed-env-name" className="app-details__selector w-200 dc__zi-12">
+            <SelectPicker
+                inputId="app-environment-select"
+                placeholder={`Select ${name}`}
+                options={options}
+                value={value}
+                onChange={onChange}
+                closeMenuOnSelect
+                isSearchable
+                classNamePrefix="app-environment-select"
+            />
+        </div>
+    </>
+)
+
+const AppSelector = ({ applications }: { applications: EnvAppsMetaDTO['apps'] }) => {
+    const { push } = useHistory()
+    const { path } = useRouteMatch()
+    const { appId, envId } = useParams<{ appId: string; envId: string }>()
+
+    const handleOnChange = (option: SelectPickerOptionType) => {
+        const newUrl = generatePath(path, { appId: option.value, envId })
+        push(newUrl)
+    }
+
+    const appIdVsNameMap = applications?.reduce((agg, curr) => {
```

Comment:
This can be null and we have missed null check below so we should do this:
```suggestion
    const appIdVsNameMap = (applications || [])?.reduce((agg, curr) => {
```